aiida_quantumespresso/cli/workflows/matdyn/base.py

```diff
@@ -12,7 +12,7 @@
 
 @cmd_launch.command('matdyn-base')
 @options.CODE(required=True, type=types.CodeParamType(entry_point='quantumespresso.matdyn'))
-@options.CALCULATION(type=types.CalculationParamType(sub_classes=('aiida.calculations:quantumespresso.q2r',)))
+@options.CALCULATION()
```

Comment:
This should actually be:
```suggestion
@options.CALCULATION(required=True)
```
Also, I understand why you removed the type restriction (it does not work), but ideally we should check that the `process_type` of the CalcJobNode corresponds to  `Q2rCalculation`, either via a string comparison to the entry point `aiida.calculations:quantumespresso.q2r` or by actually loading it and checking the class object. However, if you want to defer this, that's fine.

Reply:
Right, I will add the check in the function body for now, thanks

Reply:
Awesome, thanks!